File: app/src/main/java/ru/vtb/ppcd/example/processor/impl/BasicMessageProcessor.java
package ru.vtb.ppcd.example.processor.impl;

import ru.vtb.ppcd.example.entity.RequestEntity;
import ru.vtb.ppcd.example.processor.ExchangeContext;
import ru.vtb.ppcd.example.processor.MessageProcessor;
import ru.vtb.ppcd.example.repository.RequestRepository;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import ru.vtb.ppcd.generated.async.model.CreateProductRequestInnerEvent;

import java.util.UUID;


@Slf4j
@Service
@AllArgsConstructor
public class BasicMessageProcessor implements MessageProcessor<CreateProductRequestInnerEvent> {
    private final RequestRepository requestRepository;
    private final ObjectMapper objectMapper;

    @Override
    public void processMessage(ExchangeContext<CreateProductRequestInnerEvent> exchangeContext) {
        var message = exchangeContext.getMessage();

        RequestEntity productRequest = RequestEntity.builder()
                .initialDate(message.getTimestamp())
                .sum(message.getBody().getSum())
                .requestId(UUID.fromString(message.getId()))
                .build();


        requestRepository.save(productRequest);
    }

    @Override
    public ObjectMapper getObjectMapper() {
        return objectMapper;
    }

    @Override
    public Class<CreateProductRequestInnerEvent> processClass() {
        return CreateProductRequestInnerEvent.class;
    }
}

File: ms-ppcd-example-app/src/main/kotlin/ru/vtb/ppcd/example/processor/impl/BasicMessageProcessor.kt
package org.example.ru.vtb.ppcd.example.processor.impl

import com.fasterxml.jackson.databind.ObjectMapper
import java.util.UUID
import org.example.ru.vtb.ppcd.example.entity.RequestEntity
import org.example.ru.vtb.ppcd.example.model.ExchangeContext
import org.example.ru.vtb.ppcd.example.processor.MessageProcessor
import org.example.ru.vtb.ppcd.example.repository.RequestRepository
import org.springframework.stereotype.Component
import ru.vtb.ppcd.generated.async.model.CreateProductRequestInnerEvent


@Component
class BasicMessageProcessor(
    private val objectMapper: ObjectMapper,
    private val requestRepository: RequestRepository
) : MessageProcessor<CreateProductRequestInnerEvent> {

    override fun processMessage(exchangeContext: ExchangeContext<CreateProductRequestInnerEvent>) {
        exchangeContext.message
            .let {
                RequestEntity(
                    initialDate = it.timestamp,
                    sum = it.body.sum,
                    requestId = UUID.fromString(it.id)
                )
            }.apply { requestRepository.save(this) }
    }

    override fun getObjectMapper() = objectMapper

    override fun processClass() = CreateProductRequestInnerEvent::class.java
}